app/src/main/java/de/haw_hamburg/sensorapp/CompassFragment.java

```diff
@@ -1,23 +1,237 @@
+
+            }
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+
+            }
+        });
+    }
+
+    public void onRadioButtonClicked(View view){
+        boolean checked = ((RadioButton)view).isChecked();
+        switch (view.getId()){
+            case R.id.hardwareSensorButton:
+                if(checked){
+                    seekBar.setEnabled(true);
+                    seekBar.setProgress(compass.getLowPassFilter());
+                    progressTextView.setText("LowPassFilter: "+seekBar.getProgress()+"%");
+                    compass.setOption(1);
+                }
+                break;
+            case R.id.softwareSensorButton:
+                if(checked){
+                    seekBar.setEnabled(false);
+                    progressTextView.setText("");
+                    compass.setOption(2);
+                }
+                break;
+        }
+    }
+
+    @Override
+    public void onPause(){
+        compass.stop();
+        super.onPause();
+    }
+
+    @Override
+    public void onStart(){
+        super.onStart();
+        compass.start();
+    }
+
+    @Override
+    public void onResume(){
+        super.onResume();
+        compass.start();
+    }
+
+    @Override
+    public void onStop(){
+        compass.stop();
+        super.onStop();
+    }
+
+    class Compass implements SensorEventListener{
+        private SensorManager sensorManager;
+        private Sensor gravitiySensor;
+        private Sensor magneticSensor;
+        private Sensor rotationVectorSensor;
+        private float[] currentGravityValues = new float[3];
+        private float[] currentMagneticValues = new float[3];
+        private float azimuth = 0;
+        private float currentAzimuth = 0;
+        private ImageView compassView = null;
+        private int option;
+        private float lowPassFilter;
+
+        public Compass(Context context){
+            sensorManager = (SensorManager) context.getSystemService(Context.SENSOR_SERVICE);
+            gravitiySensor = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+            magneticSensor = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+            rotationVectorSensor = sensorManager.getDefaultSensor(Sensor.TYPE_ROTATION_VECTOR);
+            option = 2;
+            lowPassFilter = 0.95f;
+        }
+
+        public void start(){
+            sensorManager.registerListener(this, rotationVectorSensor, SensorManager.SENSOR_DELAY_GAME);
+            sensorManager.registerListener(this, gravitiySensor, SensorManager.SENSOR_DELAY_GAME);
+            sensorManager.registerListener(this, magneticSensor, SensorManager.SENSOR_DELAY_GAME);
+        }
+
+        public void stop(){
+            sensorManager.unregisterListener(this);
+        }
+
+        public void setOption(int i){
+            option = i;
+        }
+
+        public void setLowPassFilter(int i){
+            lowPassFilter = (float)i/100;
```

Comment:
This looks strange.. don't change the input paramter...make sure that the caller pass the correct value.